packages/flutter/test/material/buttons_test.dart

```diff
@@ -1195,4 +1196,123 @@ void main() {
     final Material material = tester.widget<Material>(rawButtonMaterial);
     expect(material.color, const Color(0xff00ff00));
   });
+
+  testWidgets('MaterialButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: MaterialButton(
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final MaterialButton button = tester.widget<MaterialButton>(find.byType(MaterialButton));
+    expect(button.enabled, true);
+  });
+
+  testWidgets('FlatButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: FlatButton(
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final FlatButton button = tester.widget<FlatButton>(find.byType(FlatButton));
+    expect(button.enabled, true);
+  });
+
+  testWidgets('OutlineButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: OutlineButton(
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final OutlineButton button = tester.widget<OutlineButton>(find.byType(OutlineButton));
+    expect(button.enabled, true);
+  });
+
+  testWidgets('RaisedButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: RaisedButton(
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final RaisedButton button = tester.widget<RaisedButton>(find.byType(RaisedButton));
+    expect(button.enabled, true);
+  });
+
+  testWidgets('RawMaterialButton should be enabled when onLongPress is not null.', (WidgetTester tester) async {
+    await tester.pumpWidget(
+      Directionality(
+        textDirection: TextDirection.ltr,
+        child: RawMaterialButton(
+          onPressed: null,
+          onLongPress: () { /* to make sure the button is enabled */ },
+          child: const Text('button'),
+        ),
+      ),
+    );
+
+    final RawMaterialButton button = tester.widget<RawMaterialButton>(find.byType(RawMaterialButton));
+    expect(button.enabled, true);
+  });
+
+  testWidgets('MaterialButton onPressed callback', (WidgetTester tester) async {
```

Comment:
If `onPress` callbacks of the other buttons haven't been tested, we should test them.

Reply:
Ok, so do I add another 4 tests? Or even more? (onPressed true/onLongPress false, onPressed false/OnLongPress false, and so on)? I agree with you that file is a mess.